src/printer-postcss.js

```diff
@@ -420,8 +438,14 @@ function printValue(value) {
 
 const STRING_REGEX = /(['"])(?:(?!\1)[^\\]|\\[\s\S])*\1/g;
 const NUMBER_REGEX = /(?:\d*\.\d+|\d+\.?)(?:[eE][+-]?\d+)?/g;
-const STRING_OR_NUMBER_REGEX = RegExp(
-  `${STRING_REGEX.source}|(${NUMBER_REGEX.source})`,
+const STANDARD_UNIT_REGEX = /[a-zA-Z]+/g;
+const WORD_PART_REGEX = /[$@]?[a-zA-Z_\u0080-\uFFFF][\w\-\u0080-\uFFFF]*/g;
+const ADJUST_NUMBERS_REGEX = RegExp(
+  STRING_REGEX.source +
+    `|` +
+    `(${WORD_PART_REGEX.source})?` +
+    `(${NUMBER_REGEX.source})` +
+    `(${STANDARD_UNIT_REGEX.source})?`,
```

Comment:
Who needs parsers and ASTs when you have regex?

Reply:
What's `WORD_PART_REGEX` capturing?

Reply:
When looking for a number, say `12`, there’s a risk that we find a `12` inside some identifier, such as `$myVar12`. So `WORD_PART_REGEX` matches `$myVar` in this case. In `adjustNumbers` below, the match is returned unchanged if there’s a word part in front of the number.

Reply:
`\u0080-\uFFFF` matches any non-ASCII character, because any non-ASCII character is actually valid in an `<ident-token>`: https://www.w3.org/TR/css-syntax-3/#ident-token-diagram